### DISCUSSION TITLE
Calculate staleTime / cacheTime based on response

### DISCUSSION THREAD

**Discussion by fubhy** [TanStack/query#869] [General]
I'd like to define the staleTime of a query based on its previous response. In my case, the server knows pretty accurately when data will become stale or need refetching/revalidation in the future. This is reflected both in the response header and the response payload of the query. I'd like to be able to define a callback instead of a static number to extract this information from the last query.

Is this something you'd want to put on your roadmap and consider as a feature for the next version? If so, I'd be happy to work on this and submit a PR.

**Comment by jackmellis** [comment]
Interesting idea! I think there are quite a few config options where the response is better-placed to determine the value.
This sort of thing is already available for the `retry` option as you can provide a callback that contains the failed response object.
